tools/codegen/api/python.py

```diff
@@ -173,6 +173,53 @@
 #      }
 #
 
+# TODO: stick this more firmly in the data model somewhere?
```

Comment:
nit: maybe move the helper methods to the big section below
```
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
#
#                          Python Signature
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
```

or maybe create a dedicated 'python interface' section for this (and other pyi methods).